pkg/skaffold/deploy/deploy_mux.go

```diff
@@ -97,7 +108,13 @@ func (m DeployerMux) Deploy(ctx context.Context, w io.Writer, as []graph.Artifac
 			return nil, err
 		}
 		seenNamespaces.Insert(namespaces...)
-
+		if m.iterativeStatusCheck {
+			if err = deployer.GetStatusMonitor().Check(ctx, w); err != nil {
```

Comment:
if I'm reading this right, this will return an error as soon as the first health check fails, which I think means any other errors/failures encountered here will get ignored. I wonder if it makes sense to aggregate failures here? or would that defeat the purpose of the iterative check...

Reply:
ya, the purpose of the iterative check is to fail subsequent deployments if current deployment doesn't pass the status check.